src/MotionControl.c

```diff
@@ -1355,23 +1355,25 @@ static STATUS Ros_Controller_DisableEcoMode()
 //
 // NOTE: only attempts to start job if necessary, does not reset errors, alarms.
 //       Does attempt to enable servo power (if not on)
-//       Does attempt to set the cycle mode to continuous (if not set)
+//       Does attempt to set the cycle mode to AUTO (if not set)
 //-----------------------------------------------------------------------
-BOOL Ros_MotionControl_StartMotionMode(MOTION_MODE mode)
+MotionNotReadyCode Ros_MotionControl_StartMotionMode(MOTION_MODE mode, rosidl_runtime_c__String *responseMessage)
 {
     int ret;
     MP_STD_RSP_DATA rData;
     MP_START_JOB_SEND_DATA sStartData;
     int checkCount;
     int grpNo;
+    MotionNotReadyCode motion_readiness_code = MOTION_NOT_READY_UNSPECIFIED;
+    char output[256] = { 0 };
```

Comment:
The `256` needs to be a named constant. It is used in multiple places below.

Additionally, I'm not sure that the `output` buffer is really needed. If things are checked in the correct order, then `Ros_ErrorHandling_MotionNotReadyCode_ToString` should contain sufficient information to the user _(I think)_. In that case, you shouldn't need to pass in the pointer to `responseMessage`.

Do you have some example strings from testing that you can post?